### PR TITLE
Add combined accuracy meters

### LINKED ISSUE #7292: Combined error and colour score meter
**Describe the new feature:** While one of the score meters from stable has been added, the other one hasn't. I suggest we change that. Here's a suggestion for a more modern-looking behaviour, though: instead of having the newest element scroll through the entire bar and go back to the beginning and start again, how about always having the topmost indicator be the most recent one, and with each new hit, the others would scroll down a spot (with a fluid animation) with the bottom-most one disappearing?
I would also like to take this chance to suggest an option to have both score meters displayed at the same time.

**Proposal designs of the feature:** 

image

### PR DESCRIPTION
**Pull request by mcendu**
Resolves #7292

### CODE REVIEW

**Comment by EVAST9919** [comment]
As I mentioned in https://github.com/ppy/osu/pull/7317 it would be better to make two separate dropdowns: one for position, another for type.
Like "position: left, right, (top, bottom in the future)" and "type: HitError, Colour, Combined"
(just IMO)

**Comment by mcendu** [comment]
I am not sure how to automatically convert current selection to other schemes.

**Comment by EVAST9919** [comment]
> I am not sure how to automatically convert current selection to other schemes.

Well, this can be made in a separate pr then, so no worries

File: osu.Game/Screens/Play/HUD/HitErrorDisplay.cs [hunk 1]
```diff
@@ -116,12 +153,21 @@ private void createColour(bool rightAligned)
...
+        }
+
+        private void createCombined(bool rightAligned)
+        {
+            createBar(rightAligned);
+            createColour(rightAligned);
         }
 
-        private void completeDisplayLoading(HitErrorMeter display)
+        private void completeDisplayLoading(HitErrorMeter display, bool rightAligned)
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/HitErrorDisplay.cs]: `rightAligned` seems not too future-proof seeing that there are already requests for the meters to come back as bottom-center aligned (#5932). Enum would be more readable, too.

File: osu.Game/Screens/Play/HUD/HitErrorDisplay.cs [hunk 2]
```diff
@@ -53,12 +74,15 @@ private void onNewJudgement(JudgementResult result)
                 return;
 
             foreach (var c in Children)
-                c.OnNewJudgement(result);
+            {
+                foreach (var meter in c.Children)
+                    meter.OnNewJudgement(result);
+            }
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/HitErrorDisplay.cs]: I'm not too hot on this. Can we not reuse meter instances on both sides and just update the single ones?

**Review by bdach — COMMENTED** [review]
I feel like this is a precarious piece of code that could get ugly. Maybe we want something like:
```csharp
[Flags]
enum ScoreMeterType
{
    // types
    None = 0,
    ErrorBar = 1 << 0,
    Colour = 1 << 1,
    // space for other types

    // locations - starting shift offset arbitrary
    Left = 1 << 8,
    Right = 1 << 9,
    Bottom = 1 << 10
}
```
so then you can inspect the meter type flags, create the meter instances, then inspect the meter location flags, add them wherever necessary, and done. (Or even two separate enums, although I'm not sure that's not *overly* verbose) I'm slightly concerned this is going to become switch city otherwise.

**Comment by bdach** [comment]
Oh, and tests may be nice. They're not going to look too clever (probably going to boil down to inspecting if the drawables are where they need to), but who knows what can regress in the future.

**Comment by peppy** [comment]
Alignment of these elements (and all UI elements) will be handled by a higher level component as part of the skinning/customisation engine. For now, positioning of individual elements should not be considered or implemented. Always having these drawn at the same location is fine for now.

**Comment by mcendu** [comment]
> I feel like this is a precarious piece of code that could get ugly. Maybe we want something like:
> 
> ```cs
> [Flags]
> enum ScoreMeterType
> {
>     // types
>     None = 0,
>     ErrorBar = 1 << 0,
>     Colour = 1 << 1,
>     // space for other types
> 
>     // locations - starting shift offset arbitrary
>     Left = 1 << 8,
>     Right = 1 << 9,
>     Bottom = 1 << 10
> }
> ```
> 
> so then you can inspect the meter type flags, create the meter instances, then inspect the meter location flags, add them wherever necessary, and done. (Or even two separate enums, although I'm not sure that's not _overly_ verbose) I'm slightly concerned this is going to become switch city otherwise.

My concern here is that we need to exclude the individual flags from being listed by the options, and I am not sure whether there are any means to achieve this.

File: osu.Game/Screens/Play/HUD/HitErrorDisplay.cs (hunk 2 above)
Comment (mcendu) [reply osu.Game/Screens/Play/HUD/HitErrorDisplay.cs]: IMO this looks readable and can avoid duplicating code

**Review by bdach — COMMENTED** [review]
If we're not considering positioning, maybe we should just axe the error bar location settings now and just stay with the types for the foreseeable future?

File: osu.Game/Screens/Play/HUD/HitErrorDisplay.cs (hunk 2 above)
Comment (bdach) [reply osu.Game/Screens/Play/HUD/HitErrorDisplay.cs]: I meant the double update - if you have one meter type on both sides you get two instances that update independently. Could be nice to have just one, but displayed in two places, but that's a nitpick really.

**Comment by peppy** [comment]
We'll probably still need the existing logic to some degree... I think?

This is a weird case because it isn't just repositioning but also duplicating, which is something i'm not sure we will support in the UI editor, but maybe we can/should?

**Comment by smoogipoo** [comment]
The settings have been removed, and this is now supported by the skin editor since https://github.com/ppy/osu/pull/12845

**Pull request closed without merging**